src/browser/pages/vscode.ts

```diff
@@ -5,18 +5,54 @@ const options = getOptions()
 // TODO: Add proper types.
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
-let nlsConfig: any
+export const nlsConfigElementId = "vscode-remote-nls-configuration"
+
+type NlsConfiguration = {
+  locale: string
+  availableLanguages: { [key: string]: string } | {}
+  _languagePackId?: string
+  _translationsConfigFile?: string
+  _cacheRoot?: string
+  _resolvedLanguagePackCoreLocation?: string
+  _corruptedFile?: string
+  _languagePackSupport?: boolean
+  loadBundle?: any
+}
+
+/**
+ * A helper function to get the NLS Configuration settings.
+ *
+ * This is used by VSCode for localizations (i.e. changing
+ * the display language).
+ **/
+export function getNlsConfiguration(document: Document) {
+  const nlsConfigElement = document?.getElementById(nlsConfigElementId)
+  const nlsConfig = nlsConfigElement?.getAttribute("data-settings")
+
+  if (!document || !nlsConfigElement || !nlsConfig) {
+    return undefined
+  }
+
+  try {
+    return JSON.parse(nlsConfig) as NlsConfiguration
+  } catch (error) {
+    console.error("[vscode] Failed to read or parse the NLS configuration.", error)
```

Comment:
I had originally used our `logger` but I know there was a previous discussion about using `console` for logging on the browser so that's what I did.

Plus, it's harder to mock and test our `logger`. I couldn't make it work consistently.

Reply:
If we go with throwing errors for missing elements we could just let this throw bubble up and log it elsewhere.

Yeah we have so little logging on the frontend it doesn't really seem worth bundling the logger just for one or two logging statements.

Out of curiosity, what problems did you run into testing `logger` that were resolved by switching to `console`?

Reply:
> If we go with throwing errors for missing elements we could just let this throw bubble up and log it elsewhere.

Good idea! That's easy to test too.

> Out of curiosity, what problems did you run into testing logger that were resolved by switching to console?

When I tried mocking the `logger`, I couldn't get it to work. I think it's because it will only work if you mock the `logger` then `require` the file and then use the function from the `require` import.

When I used `console`, I could overwrite it to mock it and it worked